Projects/My_website/myWebsite/blog/views.py

```diff
@@ -26,16 +26,58 @@ class userBlog(APIView):
     authentication_classes = [TokenAuthentication]
     permission_classes = [IsAuthenticated]
 
+    def get_item(self,slug):
+        item = Post.objects.get(slug=slug)
+        if item:
+            return item
+        else:
+            return False
+        
     def get(self,request):
         posts = Post.objects.filter(author=request.user.id)
         serializers = blogSerializer(posts,many=True)
         return Response(serializers.data)
     
-    def put(self,request):
-        return Response({'message':"this is put method (under devlopment)"})
-    def delete(self,request):
-        return Response({'message':"this is get method (under devlopment)"})   
-    def patch(self,request):
-        return ({'message':"this is patch method (under devlopment)"})
+    def post(self,request):
+        data = request.data
+        data['author']=request.user.id
+        serializers = blogSerializer(data=data)
+        if serializers.is_valid():
+            serializers.save()
+            return Response(serializers.data)
+        return Response(serializers.errors)
+    
+    def put(self,request,slug):
+        data = request.data
+        if data['author'] != request.user.id:
+            return Response({'message':"you are not valid author"})
```

Comment:
@codal-uchoudhary , what are we doing here ? 
You need to check whether the blog that is being edited is of the same user that is currently logged in.

we do not need to pass author key in the EDIT request payload. 

same mistake is in the patch method as well --> line 64 & 65.

Reply:
@arvish-codal  , here i am comparing the Author id of the blog  with current user id . to check the author is valid or not.

Reply:
if it return True then , the author is the right person to edit this blog.

Reply:
I tested PUT,PATCH,DELETE APIs in the postman .

Reply:
@codal-uchoudhary , yeah correct, that's what we need to do. But i think you're also passing author key in the payload. We should not be doing that.

the flow should be :
  --> there should be blog ID in the PUT request endpoint : example /blog/{id} --> /blog/1
  --> now from the Blog ID, you should be fetching the Blog instance.
  --> From the received blog instance, compare the blog's author ID with the request user id. 
  
 This is how we AUTHORISE whether the blog being edited / deleted is posted by the logged in user or not.

Reply:
okay ! let me change the methods

Reply:
@arvish-codal  , changed as per comment